x-pack/plugins/observability_onboarding/server/routes/custom_logs/route.ts

```diff
@@ -50,7 +49,9 @@ const installShipperSetupRoute = createObservabilityOnboardingServerRoute({
     const coreStart = await core.start();
 
     const kibanaUrl =
-      plugins.cloud?.setup?.kibanaUrl ?? getFallbackUrls(coreStart).kibanaUrl;
+      core.setup.http.basePath.publicBaseUrl ?? // priority given to server.publicBaseUrl
+      plugins.cloud?.setup?.kibanaUrl ?? // then cloud id
+      getFallbackUrls(coreStart).kibanaUrl; // falls back to local network binding
```

Comment:
Can you test this out in the QA environment to make sure it works in the cloud environment?

Reply:
i tested it out locally by setting `xpack.cloud.id: 'es-logs-onboarding-test:ZWFzdHV...4MjYz'`, which is the config set in cloud.

Reply:
You can create a serverless instance in the QA environment pretty easily ([pastebin](https://p.elstc.co/paste/2vxaaWso#KyfeM+ShJJwxlL6FPnMp3lWxswsdtrx-ZTTjPxNHZ/8))